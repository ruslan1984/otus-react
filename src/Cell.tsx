import React, { FC }  from 'react';


interface ICellProps {
    move?: boolean;
    onClick?: any;
    value?: string;
    disabled?: boolean;
}
interface ICellState {
    disabled: boolean;
    value: "x"|"0"|"";
}
const style = {
    width: '50px',
    height: '50px'
  };

const Presenter: FC<ICellProps> = ( { value, onClick, disabled } ) =>
    {
        return <button style={style} className="button" disabled={ disabled } onClick={ onClick }>{ value }</button>;
    }

export default class Cell extends React.Component<ICellProps, ICellState> {
    constructor ( props: ICellProps ) {
        super( props );
        this.state = {
            value: "",
            disabled: false
        };
        this.myClick = this.myClick.bind(this);
    }
    myClick() {
        const value = ( this.props.move ) ? 'x' : '0';
        this.setState( state => ( {
            value: value,
            disabled: true
        } ) );
        this.props.onClick();
    }
    render () {
        return ( //<button className="button" disabled={ this.state.disabled } onClick={ this.myClick }>{ this.state.value }</button>
            <Presenter    
                value={ this.state.value }
                onClick={ this.myClick }
                disabled={ this.state.disabled } />
        );
    }
}
