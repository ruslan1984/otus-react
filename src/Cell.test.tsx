import React from 'react';
import Cell from './Cell';

describe( "Has Button", () => {
  const mockFn = jest.fn();
  it( 'First click', () => {
    const cell = mount(
      <Cell status={"new"} cellWidth={50} cellHeight={50} onClick={ mockFn } move={ true } />
    );
    const btn = cell.find( 'button' );
    btn.simulate( 'click' );
    expect( mockFn ).toHaveBeenCalled();
    expect( btn.text() ).toEqual( 'x' );
  } );

  it( 'Secend click', () => {
    const cell = mount(
      <Cell status={"new"} cellWidth={50} cellHeight={50} onClick={ mockFn } move={ false } />
    );
    const btn = cell.find( 'button' );
    btn.simulate( 'click' );

    expect( mockFn ).toHaveBeenCalled();
    expect( btn.text() ).toEqual( '0' );
  } );
} );


describe( "snapshot Cell", () => {
  it( 'snapshot', () => {
    expect( Cell ).toMatchSnapshot()
  } )
} );