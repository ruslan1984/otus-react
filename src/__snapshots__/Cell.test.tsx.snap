// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Cell snapshot 1`] = `[Function]`;
